crates/cxx-qt-gen/src/parser/inherit.rs

```diff
@@ -0,0 +1,272 @@
+            return Err(Error::new(
+                extern_block.abi.span(),
+                "Inherit blocks must be marked with `extern \"C++\"`",
+            ));
+        }
+
+        for item in extern_block.items {
+            match item {
+                ForeignItem::Fn(function) => {
+                    base_functions.push(function);
+                }
+                _ => {
+                    return Err(Error::new(
+                        item.span(),
+                        "Only functions are allowed in #[cxx_qt::inherit] blocks",
+                    ))
+                }
+            }
+        }
+
+        Ok(InheritMethods {
+            safety,
+            base_functions,
+        })
+    }
+}
+
+/// Describes a method found in #[cxx_qt::inherit]
+pub struct ParsedInheritedMethod {
+    /// The original [syn::ForeignItemFn] of the inherited method declaration
+    pub method: ForeignItemFn,
+    /// The type of the self argument
+    pub qobject_ident: Ident,
+    /// whether the inherited method is marked as mutable
+    pub mutable: bool,
+    /// Whether the method is safe to call.
+    pub safe: bool,
+    /// the parameters of the method, without the `self` argument
+    pub parameters: Vec<ParsedFunctionParameter>,
+    /// the name of the function in Rust, as well as C++
+    pub ident: CombinedIdent,
+}
+
+impl ParsedInheritedMethod {
+    pub fn parse(method: ForeignItemFn, safety: Safety) -> Result<Self> {
+        if safety == Safety::Unsafe && method.sig.unsafety.is_none() {
+            return Err(Error::new(
+                method.span(),
+                "Inherited methods must be marked as unsafe or wrapped in an `unsafe extern \"C++\"` block!",
+            ));
+        }
+
+        let self_receiver = foreignmod::self_type_from_foreign_fn(&method.sig)?;
+        let (qobject_ident, mutability) = types::extract_qobject_ident(&self_receiver.typ)?;
+        let mutable = mutability.is_some();
+
+        let parameters = ParsedFunctionParameter::parse_all_ignoring_receiver(&method.sig)?;
+
+        let mut ident = CombinedIdent::from_rust_function(method.sig.ident.clone());
+        for attribute in &method.attrs {
+            if !attribute.path.is_ident(&format_ident!("cxx_name")) {
+                return Err(Error::new(
+                    attribute.span(),
+                    "Unsupported attribute in #[cxx_qt::inherit]",
+                ));
+            }
+
+            let name = attribute_tokens_to_value::<LitStr>(attribute)?;
+
+            ident.cpp = format_ident!("{}", name.value());
+        }
+        let safe = method.sig.unsafety.is_none();
+
+        Ok(Self {
+            method,
+            qobject_ident,
+            mutable,
+            parameters,
+            ident,
+            safe,
+        })
+    }
+
+    /// the name of the wrapper function in C++
+    pub fn wrapper_ident(&self) -> Ident {
+        format_ident!("{}CxxqtInherit", self.ident.cpp)
```

Comment:
elsewhere we use `CxxQt` not `Cxxqt` ? (eg the threading stuff)

Reply:
Done